tests/phpunit/tests/test-widget-nav-menu.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+class Widget_Nav_Menu_Test extends PLL_UnitTestCase {
+	/**
+	 * @var string
+	 */
+	public $structure = '/%postname%/';
+
+	/**
+	 * @var PLL_REST_Request
+	 */
+	private $pll_rest;
+
+	/**
+	 * @param  WP_UnitTest_Factory $factory WP_UnitTest_Factory object.
+	 * @return void
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		parent::wpSetUpBeforeClass( $factory );
+
+		self::create_language( 'en_US' );
+		self::create_language( 'fr_FR' );
+	}
+
+	/**
+	 * @global $wp_widget_factory
+	 *
+	 * @return void
+	 */
+	public function set_up() {
+		// global $wp_widget_factory;
+
+		parent::set_up();
+
+		// We need to register the nav menu ourselves since widget globals are cleaned up in test-strings.php.
+		// @see https://github.com/polylang/polylang/blob/3.2.5/tests/phpunit/tests/test-strings.php#L24-L37.
```

Comment:
This comment may quickly become obsolete as things may change when adding new tests, I suggest something more generic if need a comment.
```suggestion
		// We need to register widgets after globals cleanup.
```